pallets/phala/src/stakepool.rs

```diff
@@ -366,24 +337,63 @@ pub mod pallet {
 		#[pallet::weight(0)]
 		pub fn start_mining(
 			origin: OriginFor<T>,
-			pool_id: u64,
+			pid: u64,
 			worker: WorkerPublicKey,
 			stake: BalanceOf<T>,
 		) -> DispatchResult {
-			panic!("unimplemented")
+			let owner = ensure_signed(origin)?;
+			let mut pool_info = Self::mining_pools(pid).ok_or(Error::<T>::PoolNotExist)?;
+			// origin must be owner of pool
+			ensure!(
+				&pool_info.owner == &owner,
+				Error::<T>::UnauthorizedPoolOwner
+			);
+			// check free stake
+			ensure!(
+				&(pool_info.total_stake - pool_info.locked_stake) >= &stake,
+				Error::<T>::InsufficientStake
+			);
+			// check wheather we have add this worker
+			ensure!(
+				pool_info.workers.contains(&worker),
+				Error::<T>::WorkerHasnotAdded
+			);
+			let miner: T::AccountId = pool_sub_account(pid, &owner, &worker);
+			<mining::pallet::Pallet<T>>::set_deposit(&miner, stake);
+			<mining::pallet::Pallet<T>>::start_mining(miner)?;
```

Comment:
If it failed, should we remove the deposit? Substrate is not like EVM, an error will not get the entire transaction rolled back.